nef-plugin/SourceEditorExtension.swift

```diff
@@ -2,15 +2,77 @@
 
 import Foundation
 import XcodeKit
+import SourceEditorModels
+
+public protocol XCSourceEditorCommandDefinition {
+    var displayName: String { get }
+    var identifierKey: String { get }
+    var editorDefinitionKey: [XCSourceEditorCommandDefinitionKey : Any] { get }
+}
+
 
 class SourceEditorExtension: NSObject, XCSourceEditorExtension {
+    var commandDefinitions: [[XCSourceEditorCommandDefinitionKey : Any]] {
+        SourceEditorExtension.commands.map(\.editorDefinitionKey)
+    }
+    
+    static var commands: [Command] = {
+        #if DEBUG
+        return debugCommandDefinitions()
+        #else
+        return appstoreCommandDefinitions()
+        #endif
+    }()
+    
+    static func debugCommandDefinitions() -> [Command] {
+        [.preferences,
+         .exportSnippet(selection: ""),
+         .exportSnippetToClipboard(selection: ""),
+         .markdownPage(playground: ""),
+         .playgroundBook(package: "")]
+    }
+    
+    static func appstoreCommandDefinitions() -> [Command] {
+        [.preferences,
+         .exportSnippet(selection: ""),
+         .exportSnippetToClipboard(selection: ""),
+         .markdownPage(playground: "")]
+    }
+}
+
+
+// MARK: - Command <XCSourceEditorCommandDefinition>
+extension SourceEditorModels.Command: XCSourceEditorCommandDefinition {
+    public var displayName: String {
+        switch self {
+        case .preferences:
+            return "Preferences"
+        case .exportSnippet:
+            return "Code selection → Image (File)"
+        case .exportSnippetToClipboard:
+            return "Code selection → Image (Clipboard)"
+        case .markdownPage:
+            return "Playground       → Markdown"
+        case .playgroundBook:
+            return "Swift Package  → Playground Book (iPad)"
```

Comment:
I suggest removing those arrows (and especially the spaces around them) as the may not display properly all times (consider different accessibility settings, for instance). Instead, proper descriptions for the actions should be provided here.